### PR TITLE
Refactor `generate-keystores` script

### LINKED ISSUE #11: [generate-keystores] Allow for more granural control over keystores
Case:
I'm working on app XYZ. I wanted to generate just one keystore with (app_name=welliba, owner=welliba, flavor=tst), but it's not possible because the scripts always generates 3 keystores and always assumes that the tst keystore has owner=LeanCode, which was not correct in my case.

It's easy to copy-paste the [`gen_key` function](https://github.com/leancodepl/mobile-tools/blob/master/bin/generate-keystores#L9-L24) to terminal and just generate the one keystore, but there should be a better way to do this.

### PR DESCRIPTION
**Pull request by bartekpacia**
Today (well, yesterday) I was working on deploying an app together with @iasiu. We used `generate-keystores` which turned out to be very inflexible.

I hope it's better documented and will be easier to use after these changes.

fix #11

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -12,7 +12,7 @@ All useful scripts gathered in one place.
 
 ### generate-keystores
 
-Generates `tst`, `prod` and `prod_upload` keystore (`.jks`) files.
+Generates `tst`, `prod_upload`, and `prod` and keystore (`.jks`) files.
```
Comment (iasiu) [inline README.md:15]: Too many ands

**Review by iasiu — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -21,16 +21,19 @@ Upload created `prod` key instead of letting Google Play create one.
...
+
 Adding `prod_upload` key needs to be done in the same transaction!
 
-![](img/generate-keystores_1.png)
-![](img/generate-keystores_2.png)
+![](img/generate-keystores_1.png) ![](img/generate-keystores_2.png)
 
 ### poeditor-download
 
+> This script is deprecated. Use [poe2arb][poe2arb] instead.
```
Comment (Albert221) [inline README.md:31]: 🎉

**Review by Albert221 — APPROVED** [review]
Also remember about updating the Formula:

https://github.com/leancodepl/homebrew-tools/blob/master/Formula/mobile-tools.rb#L10

**Comment by Albert221** [comment]
Actually, do you think moving the Formulas to this repo wouldn't be more convenient?

File: bin/generate-keystore [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+#
+# $ generate-keystore "LeanCode" "New Great App" "tst"
+#
+# $ generate-keystore "Our Awesome Client" "New Great App" "prod_upload"
+#
+# $ generate-keystore "Our Awesome Client" "New Great App" "prod"
+
+
+_usage() {
+    echo "usage: generate-keystores <owner> <app_name> <flavor>"
```
[the remark this answers is not in the document]
Comment (bartekpacia) [reply bin/generate-keystore:17]: thanks!

I searched a bit and found [this thread](https://stackoverflow.com/questions/4919093/should-i-log-messages-to-stderr-or-stdout), good read.

File: bin/generate-keystore [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+flavor="${3:-}"
+if [ -z "$flavor" ]; then
+    echo "flavor is not set"
+    _usage
+    exit 1
+fi
+
+echo "Creating keystore for app $app_name (flavor: $flavor, owner: $owner) ..."
+
+keytool -genkey \
```
[the remark this answers is not in the document]
Comment (bartekpacia) [reply bin/generate-keystore:47]: What's the gain in replacing the process?

File: bin/generate-keystore [hunk 5]
```diff
@@ -0,0 +1,56 @@
+#!/usr/bin/env bash
```
[the remark this answers is not in the document]
Comment (bartekpacia) [reply bin/generate-keystore:1]: I use this shebang because almost all shellscripts at LeanCode use it.

**Pull request merged**